### PR TITLE
Adding EnableSqlCommandTextInstrumentation to ApplicationInsightsLoggerOptions

### LINKED ISSUE #2548: Add ability to turn on SQL query logging for Azure Functions
Before 28 july 2020 we were able to see the sql queries in appinsight dependencies logs, but now we are only seeing server names and no sql queries are logged from azure functions consumption plan.  We have other code in aks cluster and sql queries are logged for calls done for app hosted in aks.

#### Expected behavior

Sql queries not logged in appinsight for azure function consumption plan.

#### Actual behavior

Sql queries should be logged in appinsight for azure function consumption plan.

#### Known workarounds
i tried this link but it didnt help
https://github.com/microsoft/ApplicationInsights-dotnet/issues/1425
Also adding reference to this dll brings aspnetcore things in azurefuctions project and is not making sense 

#### Related information 

Provide any related information 

* Package version
* Links to source
image

### PR DESCRIPTION
**Pull request by soninaren**
Resolves https://github.com/Azure/azure-webjobs-sdk/issues/2548

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLoggerOptions.cs [hunk 1]
```diff
@@ -105,6 +105,9 @@ public string QuickPulseAuthenticationApiKey
         /// </summary>
         public bool EnableDependencyTracking { get; set; } = true;
 
+
+        public bool EnableSqlCommandTextInstrumentation { get; set; } = false;
```
Comment (brettsam) [inline src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLoggerOptions.cs]: I think we should put this into a new  `DependencyTrackingOptions` type, which then lives here. It looks like there's a handful of other Enable/Disable properties on the module that we can also add to this type and just cover them all at once..

File: src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/DependencyTrackingOptions.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.WebJobs.Logging.ApplicationInsights
+{
+    public class DependencyTrackingOptions
```
Comment (brettsam) [inline src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/DependencyTrackingOptions.cs:8]: We'll need to initialize some of these that have defaults of `true` to make sure they match if someone doesn't set them. https://github.com/microsoft/ApplicationInsights-dotnet/blob/develop/WEB/Src/DependencyCollector/DependencyCollector/DependencyTrackingTelemetryModule.cs#L47-L96

File: src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLoggerOptions.cs [hunk 3]
```diff
@@ -159,6 +162,18 @@ public string Format()
                 { nameof(HttpAutoCollectionOptions.EnableResponseHeaderInjection), HttpAutoCollectionOptions.EnableResponseHeaderInjection }
             };
 
+            JObject dependencyTrackingOptions = new JObject
```
Comment (brettsam) [inline src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLoggerOptions.cs]: do we need to check if `DependencyTrackingOptions` is null here?

File: src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLoggerOptions.cs [hunk 4]
```diff
@@ -105,6 +106,8 @@ public string QuickPulseAuthenticationApiKey
         /// </summary>
         public bool EnableDependencyTracking { get; set; } = true;
 
+        public DependencyTrackingOptions DependencyTrackingOptions { get; set; }
```
Comment (brettsam) [inline src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLoggerOptions.cs:112]: Add a summary comment here. And be sure to mention that this doesn't apply if `EnableDependencyTracking` is `false`.

File: src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLoggerOptions.cs (hunk 3 above)
Comment (soninaren) [reply src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLoggerOptions.cs]: fixed

File: src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLoggerOptions.cs (hunk 4 above)
Comment (soninaren) [reply src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLoggerOptions.cs:112]: Fixed

File: src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLoggerOptions.cs [hunk 5]
```diff
@@ -105,6 +106,11 @@ public string QuickPulseAuthenticationApiKey
         /// </summary>
         public bool EnableDependencyTracking { get; set; } = true;
 
+        /// <summary>
+        /// Configuration for dependency tracking. The dependecny tracking configuration only takes effect if EnableDependencyTracking is set to true
```
Comment (brettsam) [inline src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLoggerOptions.cs:110]: nit -- spelling of `dependecny`

File: src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/DependencyTrackingOptions.cs (hunk 2 above)
Comment (brettsam) [reply src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/DependencyTrackingOptions.cs:8]: Let's add a comment that these defaults are all coming from the link above. But let's link directly to a commit like: https://github.com/microsoft/ApplicationInsights-dotnet/blob/3933341dc9759a388a0601b47b6108f8570d15b1/WEB/Src/DependencyCollector/DependencyCollector/DependencyTrackingTelemetryModule.cs

**Review by brettsam — APPROVED** [review]

**Comment by brettsam** [comment]
@soninaren -- we'll need to pull this into the host and make sure it binds as expected to host.json

**Comment by Kapkan-Lesion** [comment]
@brettsam @soninaren 
Hi! Why should we expect this to become publicly available? Am desperately looking for a way to turn SQL commands back on in my Azure Function's App Insights!

**Comment by brettsam** [comment]
It will be part of the next Functions release. Likely worldwide in a few weeks.

**Comment by Kapkan-Lesion** [comment]
@brettsam Hey Brett, apologies for chasing this feature, when will this feature become available? I understand that you guys have a lot going on especially during this pandemic, just wanted to say thanks for the hard work!

**Comment by soninaren** [comment]
@Kapkan-Lesion This should be available for function runtime version 3.0.15027 or higher.

**Pull request merged**